### PR TITLE
[1.14] Configuring non trivial KafkaChannel options

### PR DESCRIPTION
**Pull request by cardil**
[1.14] Apart from adding retries, this configures also the number of partitions and replication factor. Both of those is 1 by default.

This PR should address https://github.com/knative-sandbox/eventing-kafka/issues/533 for serverless-operator and productized builds.

### CODE REVIEW

File: hack/lib/serverless.bash [hunk 1]
```diff
@@ -218,16 +218,19 @@ EOF
 
 function ensure_kafka_channel_default {
   logger.info 'Set KafkaChannel as default'
-
+  local defaultChConfig channelTemplateSpec yamls
+  yamls="$(dirname "$(realpath "${BASH_SOURCE[0]}")")/yamls"
+  read.in defaultChConfig < "${yamls}/kafka-default-ch-config.yaml"
+  read.in channelTemplateSpec < "${yamls}/kafka-channel-templatespec.yaml"
```
Comment (markusthoemmes) [inline hack/lib/serverless.bash:224]: I read the comment on the `read.in` function but honestly don't quite understand the reasoning :sweat_smile:. What's the benefit vs. doing `defaultChConfig=$(cat "${yamls}/kafka-default-ch-config.yaml)`?

Comment (cardil) [reply hack/lib/serverless.bash:224]: You're right. Let's simplify that. I just thought it will be necessary...

**Comment by cardil** [comment]
Flake `TestContainerSourceV1Alpha2`

/test 4.7-upstream-e2e-aws-ocp-47

**Pull request merged**